Code/GraphMol/QueryOps.h

```diff
@@ -922,12 +924,86 @@ class HasPropWithValueQuery<TargetPtr, std::string>
   }
 };
 
+template <class TargetPtr>
+class HasPropWithValueQuery<TargetPtr, ExplicitBitVect>
+    : public Queries::EqualityQuery<int, TargetPtr, true> {
+  std::string propname;
+  ExplicitBitVect val;
+  float tol;
+
+ public:
+  HasPropWithValueQuery()
+      : Queries::EqualityQuery<int, TargetPtr, true>(), propname(), val(), tol(1.0) {
+    this->setDescription("HasPropWithValue");
+    this->setDataFunc(0);
+  };
+  
+  explicit HasPropWithValueQuery(const std::string &prop,
+                                 const ExplicitBitVect &v,
+                                 float tol = 1.0)
+      : Queries::EqualityQuery<int, TargetPtr, true>(), propname(prop), val(v), tol(tol) {
+    this->setDescription("HasPropWithValue");
+    this->setDataFunc(0);
+  };
+
+  virtual bool Match(const TargetPtr what) const {
+    bool res = what->hasProp(propname);
+    if (res) {
+      try {
+        const ExplicitBitVect &bv = what->template getProp<const ExplicitBitVect&>(propname);
```

Comment:
this use of `template` is, IMO, a really horrible bit of C++ syntax. Is it actually necessary here?

Reply:
I think this was left over from a really old compiler.  I’m happy to take it out.

Reply:
Unlucky... ah well, guess we're stuck with it

Reply:
I think this happens because `what` is also a template and during the first pass, the compiler doesn't know the type.